src/js/components/Box/doc.js

```diff
@@ -127,12 +127,19 @@ export const doc = Box => {
       `Include a border. 'between' will place a border in the gap between
       child elements. You must have a 'gap' to use 'between'.`,
     ),
-    direction: PropTypes.oneOf([
-      'row',
-      'column',
-      'row-responsive',
-      'row-reverse',
-      'column-reverse',
+    direction: PropTypes.oneOfType([
+      PropTypes.oneOf([
+        'row',
+        'column',
+        'row-responsive',
+        'row-reverse',
+        'column-reverse',
+      ]),
+      PropTypes.shape({
+        direction: PropTypes.string,
+        responsive: PropTypes.bool,
+        reverse: PropTypes.bool,
```

Comment:
If I'm understanding the intent, that `reverse` only applies when `responsive` is true, I'm wondering if we should structure this object a bit differently to be clearer. For example:
`direction={{ direction: 'row', responsive: { direction: 'column-reverse } }}`